### PR TITLE
498 bug rainbow crack

### PR DESCRIPTION
**Pull request by Siddh2001**
**Title: Update RainbowCrack.tsx Component with Accurate Descriptions and Steps**

**Description:**  This pull request updates the RainbowCrack.tsx component with more accurate and detailed descriptions, usage steps and dependencies. The changes aim to improve the clarity and usability of the tool for end users.

**Description:** Updated the component description to better reflect the functionality of RainbowCrack as a command-line tool that uses rainbow tables to crack various password hashes (LM, NTLM, MD5, SHA1, SHA256).

**Steps:** Enhanced the step-by-step guide for using the tool, including specific command examples for cracking single and multiple hashes, as well as LM and NTLM hashes from pwdump files. Provided clear instructions for storing rainbow tables and executing the crack operation.

**Dependencies:** Updated the dependencies to correctly reflect the required command (rcrack).

**Reason for Change:** The previous tooltip and steps were inaccurate and did not align with the actual functionality of RainbowCrack. The updated descriptions and instructions provide a clearer, more user-friendly guide for utilizing the tool effectively.

**Testing:** Verified that the updated descriptions and steps render correctly in the UI. Ensured that the dependency update reflects the accurate command required for execution.

**Related Issues:** None.

**Reviewer Notes:** Please review the updated descriptions and steps to ensure they align with the intended usage of the tool.
Any feedback on further enhancements or corrections is appreciated.

### CODE REVIEW

File: src/components/Rainbowcrack/Rainbowcrack.tsx [hunk 1]
```diff
@@ -32,17 +32,26 @@ const RainbowCrack = () => {
...
+        "1. Ensure that your rainbow tables (*.rt, *.rtc) are stored in a directory.\n" +
+        "2. Enter the hash value you want to crack in the input field. For a single hash, use the format: ./rcrack [path to tables] -h [hash].\n" +
+        "   - Example: ./rcrack . -h 5d41402abc4b2a76b9719d911017c592\n" +
+        "3. To crack multiple hashes from a file, use the format: ./rcrack [path to tables] -l [hash_list_file].\n" +
+        "   - Example: ./rcrack . -l hash.txt\n" +
+        "4. If you have LM hashes in a pwdump file, use the format: ./rcrack [path to tables] -lm [pwdump_file].\n" +
+        "   - Example: ./rcrack . -lm pwdump.txt\n" +
+        "5. For NTLM hashes in a pwdump file, use the format: ./rcrack [path to tables] -ntlm [pwdump_file].\n" +
+        "   - Example: ./rcrack . -ntlm pwdump.txt\n" +
+        "6. Click the 'Crack' button to execute the command and display the results."; // Steps to use the component.
```
Comment (michaeljpigott) [inline src/components/Rainbowcrack/Rainbowcrack.tsx:51]: The format for this should be Start + Title
```suggestion
        "6. Click the Start Crack " +
        title + 
        " button to execute the command and display the results."; // Steps to use the component.
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Siddh2001 

I think instead of having a workaround in the user guide, it would better to fix this tool. 
1. The command should be changed from rcrack to ./rcrack because that is what is used in the command line. 
2. "-h" needs to be added to the variable args. 
3.  Potentially you also need to add "." to the variable args as well for the path. 
4. You need to update the description of this tool in RouteWrapper.txt because it says that Rainbowcrack is used to generate rainbow tables, where this particular tool is using rainbow tables to crack hashes. 

Interestingly, the tool did not work until I sorted the rainbow table using rtsort. Therefore, I think is potential adding rtsort to this tool as well to make it more useful. 

Kind regards

Michael

**Comment by michaeljpigott** [comment]
Hi @Siddh2001 

I tested it and found the command rcrack works just fine rather than ./rcrack:

image

File: src/components/Rainbowcrack/Rainbowcrack.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Siddh2001) [reply src/components/Rainbowcrack/Rainbowcrack.tsx:51]: I don't think its a good idea to include the title 
the statement becomes 
"Click the Crack RainbowCrack button to execute the command and display the results."

rather we could go for 
Click the Crack button to execute the command and display the results.

Comment (Siddh2001) [reply src/components/Rainbowcrack/Rainbowcrack.tsx:51]: I don't think its a good idea to include the title
the statement becomes
"Click the Crack RainbowCrack button to execute the command and display the results."

rather we could go for
Click the Crack button to execute the command and display the results

Comment (michaeljpigott) [reply src/components/Rainbowcrack/Rainbowcrack.tsx]: Should be "Click the Start RainbowCrack button" to match our convention, but it is a minor thing, so no worries.

**Review by michaeljpigott — APPROVED** [review]
Hi @Siddh2001 

Thank you for making those changes. I approve this pull request. 

Please note that I think this tool has potential to be improved, e.g. add a dropdown list to select either the -h, -l, -lm, etc parameters. 

I also could not get the tool to work: 

image

Not sure what the issue is there. I was following the steps from [this guide](https://medium.com/@tech_with_orgito_/rainbow-tables-rainbowcrack-cracking-passwords-on-kali-linux-5800e7048526). 

Kind regards

Michael

**Comment by Siddh2001** [comment]
I am well aware that this tools needs a lot of features to be implemented to work and dependencies as well, we can discuss it and create a new issue for the new students working on the project next trimester.
This issue was particularly based on bugs so might have to create a new one which is more precise on what needs to be done

**Comment by michaeljpigott** [comment]
Sounds good. 😊

**Pull request merged**